packages/api/src/controllers/asset.ts

```diff
@@ -527,23 +536,90 @@ app.post(
       asset
     );
 
-    res.json({ url, asset, task });
+    res.json({ url, tusEndpoint, asset, task });
   }
 );
 
-app.put("/upload/:url", async (req, res) => {
-  const {
-    params: { url },
-    config: { jwtSecret, jwtAudience },
-  } = req;
-  const { uploadUrl, playbackId } = parseUploadUrl(url, jwtSecret, jwtAudience);
+let tusServer: tus.Server;
 
-  const assets = await db.asset.find({ playbackId }, { useReplica: false });
-  if (!assets?.length || !assets[0]?.length) {
-    throw new NotFoundError(`asset not found`);
+export const setupTus = async (objectStoreId: string): Promise<void> => {
+  tusServer = await createTusServer(objectStoreId);
+};
+
+async function createTusServer(objectStoreId: string) {
+  const os = await db.objectStore.get(objectStoreId);
+
+  const url = new URL(os.url);
+  const [_, vodRegion, vodBucket] = url.pathname.split("/");
+  let protocol = url.protocol;
+  if (protocol.includes("+")) {
+    protocol = protocol.split("+")[1];
+  }
+
+  const opts: tus.S3StoreOptions | S3ClientConfig = {
+    path: "/upload/tus",
+    bucket: vodBucket,
+    accessKeyId: url.username,
+    secretAccessKey: url.password,
+    region: vodRegion,
+    partSize: 8 * 1024 * 1024,
+    tmpDirPrefix: "tus-tmp-files",
+    endpoint: "https://storage.googleapis.com",
```

Comment:
This should actually come from the parsed `url` above, it doesn't have to/shouldn't be hardcoded. Can just follow the logic from [`getS3PresignedUrl`](https://github.com/livepeer/livepeer-com/blob/6b86fb011c94e9c6dee5c9837110e0bdf7f08663/packages/api/src/controllers/helpers.ts#L141), tho you can probably just omit the user/password from the endpoint since we are providing them raw above.

probably sth like this works
```suggestion
    endpoint: `${protocol}//${url.host}`,
```